eslint-rules/index.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable @wordpress/no-unused-vars-before-return */
+/**
+ * ESLint rules.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module.exports = {
+	rules: {
+		'camelcase-acronyms': {
+			create( context ) {
+				const options = context.options[ 0 ] || {};
+				let properties = options.properties || '';
+
+				if ( properties !== 'always' && properties !== 'never' ) {
+					properties = 'always';
+				}
+
+				//--------------------------------------------------------------------------
+				// Helpers
+				//--------------------------------------------------------------------------
+
+				// contains reported nodes to avoid reporting twice on destructuring with shorthand notation
+				const reported = [];
+				const acronyms = [
+					'AMP',
+					'HTML',
+					'ID',
+					'JSON',
+					'URL',
+				];
+
+				/**
+				 * Reports an AST node as a rule violation.
+				 *
+				 * @param {Object} node The node to report.
+				 * @return {void}
+				 * @private
+				 */
+				function report( node ) {
+					if ( ! reported.includes( node ) ) {
+						reported.push( node );
+						context.report( { node, message: `\`${ node.name }\` violates naming rules.`, data: { name: node.name } } );
+					}
+				}
+
+				return {
+					Identifier( node ) {
+						const name = node.name;
+
+						// Ignore imports, because they may not respect our rules.
+						if ( node.parent && (
+							node.parent.type === 'ImportDefaultSpecifier' ||
+							node.parent.type === 'ImportSpecifier' ||
+							( node.parent.parent && (
+								node.parent.parent.type === 'ExportDefaultDeclaration' ||
+								node.parent.parent.type === 'ExportDeclaration'
+							) )
+						) ) {
+							return;
+						}
+
+						// Ignore known, JS globals like `document` and `window`.
+						// `document.getElementById` should not set off this rule.
+						if (
+							node.parent && node.parent.object &&
+							(
+								(
+									node.parent.object.name === 'document' ||
+									node.parent.object.name === 'global' ||
+									node.parent.object.name === 'window'
+								) || (
+									node.parent.object.object && (
+										node.parent.object.object.name === 'document' ||
+										node.parent.object.object.name === 'global' ||
+										node.parent.object.object.name === 'window'
+									)
+								)
```

Comment:
Eventually adding a recursive check for these types of things would be nice, but this basic one covers all of our bases and should cover us without forcing all identifies to be recursively searched, which would be a bit more expensive. I think this is a good middle-ground between preventing false-positives and efficiency 🙂